### DISCUSSION TITLE
Integration with GPT API / similar

### DISCUSSION THREAD

**Discussion by AdvikMehta** [NeonGeckoCom/NeonCore#553] [Q&A] [answered]
Hello!

I have just recently came across Neon AI, and from what little I understand, it is a completely local assistant. Please forgive my lack of knowledge, but I was wondering how one can integrate the GPT API (with langchain) into it. Basically, I want Neon to use the chat gpt api if none of the other skills are triggered by the prompt. Moreover, I want my assistant to sound a certain way (e.g. like Snoop Dogg [[link](https://dev.to/mindsdb/build-a-celebrity-twitter-chatbot-with-gpt-4-3l34)]) in how it answers the prompt (including other skills). 

As an alternative, would I just be able to use the neon-skills repo and integrate it with my own impl for the chatbot? How would I go about it? Any help would be very much appreciated!

Best,
Advik

**Comment by JarbasAl** [comment, marked as answer]
sounds like you want to keep an eye on this issue https://github.com/OpenVoiceOS/ovos-persona/issues/4

Neon is built on top of ovos-core so this all will work under Neon when done, in the meantime Neon has a ChatGPT skill already 

https://github.com/NeonGeckoCom/skill-fallback_llm

**Reply by AdvikMehta** [reply]
Thank you for your response @JarbasAl! Looks like the ChatGPT skill is the way to go for me. 

Also, I wonder if it would be feasible to send responses from other skills to the LLM to be re-phrased as per the required persona before passing it to TTS (while the ovos issue is still open).
